src/internal/group_api/requests.rs

```diff
@@ -226,14 +227,14 @@ pub mod group_create {
 pub mod group_get {
     use super::*;
 
-    pub fn group_get_request(
-        auth: &RequestAuth,
+    pub fn group_get_request<'a>(
+        auth: &'a RequestAuth,
         id: &GroupId,
-    ) -> impl Future<Item = GroupGetApiResponse, Error = IronOxideErr> {
+    ) -> impl Future<Item = GroupGetApiResponse, Error = IronOxideErr> + 'a {
         auth.request.get(
             &format!("groups/{}", rest::url_encode(&id.0)),
             RequestErrorCode::GroupGet,
-            &auth.create_signature(Utc::now()),
+            AuthV2Builder::new(&auth, Utc::now()),
```

Comment:
Maybe -- I've thought the same thing, actually. In general I prefer to have all the state passed in for testability (in cases where we _do_ want to specify the timestamp), but that wouldn't preclude us from making a version without the timestamp. But if we did that would there be any code outside of test code using `new`? Probably not.

And so I just end up leaving it. If you feel strongly about it, we can discuss further, but otherwise I'm inclined to leave it as is.

Reply:
Probably not worth messing with - a separate version with the parameter just for testing makes me a little jumpy.